internal/fs/inode/dir.go

```diff
@@ -939,3 +942,17 @@ func (d *dirInode) ShouldInvalidateKernelListCache(ttl time.Duration) bool {
 	cachedDuration := d.cacheClock.Now().Sub(d.prevDirListingTimeStamp)
 	return cachedDuration >= ttl
 }
+
+func (d *dirInode) RenameFolder(ctx context.Context, folderName string, destinationFolderId string) (folder *gcs.Folder, err error) {
+	folder, err = d.bucket.RenameFolder(ctx, folderName, destinationFolderId)
+	if err != nil {
+		return nil, err
+	}
+
+	// Remove old entry from type cache.
```

Comment:
No need to update stat cache here ?

Reply:
Stat cache will update when we call RenameFolder method as a part of fast stat bucket wrapper.